src/wp-includes/blocks.php

```diff
@@ -331,8 +335,9 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	}
 
 	// Try to get metadata from the static cache for core blocks.
-	$metadata = false;
-	if ( str_starts_with( $file_or_folder, ABSPATH . WPINC ) ) {
+	$metadata      = false;
+	$is_core_block = str_starts_with( $file_or_folder, ABSPATH . WPINC );
```

Comment:
Should we consider a helper function since this logic is being repeated in both functions here?

Reply:
I think that is a good idea!.

Reply:
Also I do the same thing here - https://github.com/WordPress/wordpress-develop/pull/4492/files